Cannot find module '.../node_modules/@auth/core/providers/email' imported from .../node_modules/@auth/core/providers/nodemailer.js

### Provider type

Email

### Environment

  System:
    OS: Linux 6.5 Ubuntu 23.10 23.10 (Mantic Minotaur)
    CPU: (8) x64 Intel(R) Core(TM) i7-8705G CPU @ 3.10GHz
    Memory: 3.65 GB / 15.33 GB
    Container: Yes
    Shell: 5.2.15 - /bin/bash
  Binaries:
    Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
    npm: 10.2.3 - ~/.nvm/versions/node/v20.10.0/bin/npm
  npmPackages:
    @auth/core: ^0.25.0 => 0.25.0 
    @auth/sequelize-adapter: ^1.2.0 => 1.2.0 
    @auth/sveltekit: ^0.11.0 => 0.11.0 


### Reproduction URL

https://github.com/derheld42/email-auth-issue

### Describe the issue

While attempting to upgrade a project to a later @auth/core, the email Provider / Nodemailer provider no longer imports. I created a svelte project using "npm create svelte@latest my-app" and added a single line into src/hooks.server.ts to demonstrate something is amiss.

I also tried seeing if instructions were updated here: https://authjs.dev/guides/providers/email

npm run dev
` VITE v5.0.12  ready in 789 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: use --host to expose
  ➜  press h + enter to show help
7:22:56 p.m. [vite] Error when evaluating SSR module /src/hooks.server.ts: failed to import "@auth/core/providers/email"
|- Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/user/proj/email-auth-issue/node_modules/@auth/core/providers/email' imported from /home/user/proj/email-auth-issue/node_modules/@auth/core/providers/nodemailer.js
    at finalizeResolution (node:internal/modules/esm/resolve:255:11)
    at moduleResolve (node:internal/modules/esm/resolve:908:10)
    at defaultResolve (node:internal/modules/esm/resolve:1121:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:396:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:365:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:240:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:85:39)
    at link (node:internal/modules/esm/module_job:84:36)

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/user/proj/email-auth-issue/node_modules/@auth/core/providers/email' imported from /home/user/proj/email-auth-issue/node_modules/@auth/core/providers/nodemailer.js
    at finalizeResolution (node:internal/modules/esm/resolve:255:11)
    at moduleResolve (node:internal/modules/esm/resolve:908:10)
    at defaultResolve (node:internal/modules/esm/resolve:1121:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:396:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:365:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:240:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:85:39)
    at link (node:internal/modules/esm/module_job:84:36) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///home/user/proj/email-auth-issue/node_modules/@auth/core/providers/email'
}
`

### How to reproduce

npm install
npm run dev

### Expected behavior

Shouldn't see any import errors from @auth/core/provider/nodemailer or @auth/core/provider/email

--- a/packages/core/src/providers/email.ts
+++ b/packages/core/src/providers/email.ts
@@ -43,8 +43,10 @@ export interface EmailConfig extends CommonProviderOptions {
   }) => Awaitable<void>
   /** Used to hash the verification token. */
   secret?: string
-  /** Used with HTTP-based email providers  */
+  /** Used with HTTP-based email providers. */
   apiKey?: string
+  /** Used with SMTP-based email providers. */
+  server?: NodemailerConfig["server"]
   generateVerificationToken?: () => Awaitable<string>
   normalizeIdentifier?: (identifier: string) => string
   options: EmailUserConfig

--- a/packages/core/src/providers/nodemailer.ts
+++ b/packages/core/src/providers/nodemailer.ts
@@ -1,5 +1,6 @@
 import { createTransport } from "nodemailer"
-import { EmailConfig, html, text } from "./email"
+import { EmailConfig, html, text } from "./email.js"
+import { AuthError } from "../errors.js"
 
 import type { Transport, TransportOptions } from "nodemailer"
 import * as JSONTransport from "nodemailer/lib/json-transport/index.js"
@@ -8,7 +9,7 @@ import * as SESTransport from "nodemailer/lib/ses-transport/index.js"
 import * as SMTPTransport from "nodemailer/lib/smtp-transport/index.js"
 import * as SMTPPool from "nodemailer/lib/smtp-pool/index.js"
 import * as StreamTransport from "nodemailer/lib/stream-transport/index.js"
-import type { Awaitable, Theme } from "../types"
+import type { Awaitable, Theme } from "../types.js"
 
 type AllTransportOptions =
   | string
@@ -27,9 +28,8 @@ type AllTransportOptions =
   | Transport<any>
   | TransportOptions
 
-export interface NodemailerConfig
-  extends Omit<EmailConfig, "sendVerificationRequest" | "options"> {
-  server: AllTransportOptions
+export interface NodemailerConfig extends EmailConfig {
+  server?: AllTransportOptions
   sendVerificationRequest: (params: {
     identifier: string
     url: string
@@ -50,6 +50,9 @@ export type NodemailerUserConfig = Omit<
 export default function Nodemailer(
   config: NodemailerUserConfig
 ): NodemailerConfig {
+  if (!config.server)
+    throw new AuthError("Nodemailer requires a `server` configuration")
+
   return {
     id: "nodemailer",
     type: "email",

